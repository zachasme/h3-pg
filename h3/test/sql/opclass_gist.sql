\pset tuples_only on
\set hexagon '\'831c02fffffffff\'::h3index'

CREATE TABLE h3_test_gist (cell h3index);
CREATE INDEX h3_test_gist_index
          ON h3_test_gist
       USING gist(cell experimental_h3index_ops);

-- insert data
INSERT INTO h3_test_gist (cell) SELECT h3_cell_to_parent(:hexagon);
INSERT INTO h3_test_gist (cell) SELECT h3_cell_to_children(:hexagon);
INSERT INTO h3_test_gist (cell) SELECT h3_cell_to_center_child(:hexagon, 15);

--
-- TEST GiST
--
SELECT COUNT(*) = 1 FROM h3_test_gist WHERE cell @> :hexagon;
SELECT COUNT(*) = 8 FROM h3_test_gist WHERE cell <@ :hexagon;

INSERT INTO h3_test_gist (cell) SELECT h3_grid_disk(:hexagon, 2);
SELECT COUNT(*) = 8 FROM h3_test_gist WHERE cell <-> :hexagon = 2;

--
-- BREAK GIST
--

SELECT COUNT(*) = 8 FROM h3_test_gist WHERE cell <@ :hexagon;
SELECT COUNT(*) FROM h3_test_gist WHERE cell <@ :hexagon;
INSERT INTO h3_test_gist (cell) SELECT h3_cell_to_children(:hexagon, 8);
SELECT COUNT(*) > 8 FROM h3_test_gist WHERE cell <@ :hexagon;
