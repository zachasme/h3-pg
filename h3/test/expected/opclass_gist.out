\pset tuples_only on
\set hexagon '\'831c02fffffffff\'::h3index'
CREATE TABLE h3_test_gist (hex h3index);
CREATE INDEX GIST_IDX ON h3_test_gist USING gist(hex);
INSERT INTO h3_test_gist (hex) SELECT h3_cell_to_parent(:hexagon);
INSERT INTO h3_test_gist (hex) SELECT h3_cell_to_children(:hexagon);
INSERT INTO h3_test_gist (hex) SELECT h3_cell_to_center_child(:hexagon, 15);
--
-- TEST GiST
--
SELECT COUNT(*) = 1 FROM h3_test_gist WHERE hex @> :hexagon;
 t

SELECT COUNT(*) = 8 FROM h3_test_gist WHERE hex <@ :hexagon;
 t

INSERT INTO h3_test_gist (hex) SELECT h3_grid_disk(:hexagon, 2);
SELECT COUNT(*) = 8 FROM h3_test_gist WHERE hex <-> :hexagon = 2;
 t

--
-- BREAK GIST
--
SELECT COUNT(*) = 8 FROM h3_test_gist WHERE hex <@ :hexagon;
 t

SELECT COUNT(*) FROM h3_test_gist WHERE hex <@ :hexagon;
     8

INSERT INTO h3_test_gist (hex) SELECT h3_cell_to_children(:hexagon, 8);
SELECT COUNT(*) > 8 FROM h3_test_gist WHERE hex <@ :hexagon;
 t

